src/redteam/commands/discover.ts

```diff
@@ -0,0 +1,318 @@
+import chalk from 'chalk';
+import cliProgress from 'cli-progress';
+import { type Command, Option } from 'commander';
+import dedent from 'dedent';
+import * as fs from 'fs';
+import yaml from 'js-yaml';
+import path from 'path';
+import { z } from 'zod';
+import cliState from '../../cliState';
+import { VERSION } from '../../constants';
+import { fetchWithProxy } from '../../fetch';
+import { getUserEmail } from '../../globalConfig/accounts';
+import { cloudConfig } from '../../globalConfig/cloud';
+import logger, { setLogLevel } from '../../logger';
+import { loadApiProvider, loadApiProviders } from '../../providers';
+import telemetry from '../../telemetry';
+import type { ApiProvider, UnifiedConfig } from '../../types';
+import { setupEnv } from '../../util';
+import { getProviderFromCloud } from '../../util/cloud';
+import { readConfig } from '../../util/config/load';
+import { writePromptfooConfig } from '../../util/config/manage';
+import invariant from '../../util/invariant';
+import { getRemoteGenerationUrl } from '../remoteGeneration';
+
+const ArgsSchema = z
+  .object({
+    config: z.string().optional(),
+    output: z.string(),
+    target: z.string().optional(),
+    envPath: z.string().optional(),
+    verbose: z.boolean().optional(),
+    preview: z.boolean(),
+    turns: z.number().optional(),
+  })
+  .refine((data) => !(data.config && data.target), {
+    message: 'Cannot specify both config and target!',
+    path: ['config', 'target'],
+  });
+
+type Args = z.infer<typeof ArgsSchema>;
+
+const DEFAULT_OUTPUT_PATH = 'redteam.yaml';
+
+/**
+ * Queries Cloud for the purpose-discovery logic, sends each logic to the target,
+ * and summarizes the results.
+ *
+ * @param target - The target API provider.
+ * @param maxTurns - The maximum number of turns to run the discovery process.
+ * @returns The purpose of the target.
+ */
+export async function doTargetPurposeDiscovery(
```

Comment:
We need a nice behavior for if remote generation is disabled. You should talk about this in.the docs too

Reply:
Because remote generation is a necessary prerequisite, I would think the command should log a warning and return immediately